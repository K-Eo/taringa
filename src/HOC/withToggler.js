import React, { Fragment, PureComponent } from 'react'

const withToggler = Toggler => BaseComponent => {
  return class withTogglerHOC extends PureComponent {
    constructor(props) {
      super(props)

      this.state = {
        isVisible: false,
      }

      this.handleToggleVisibility = this.handleToggleVisibility.bind(this)
    }

    handleToggleVisibility(e) {
      if (e && e.preventDefault) {
        e.preventDefault()
      }

      this.setState({ isVisible: !this.state.isVisible })
    }

    render() {
      const { isVisible } = this.state

      return (
        <Fragment>
          <Toggler
            {...this.props}
            isVisible={isVisible}
            onToggleVisibility={this.handleToggleVisibility}
          />
          {isVisible && <BaseComponent {...this.props} />}
        </Fragment>
      )
    }
  }
}

export default withToggler
