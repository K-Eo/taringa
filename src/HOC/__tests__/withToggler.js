import React from 'react'

import withToggler from '../withToggler'

const BaseComponent = () => {
  return <p>BaseComponent</p>
}

const Toggler = () => {
  return <p>Toggler</p>
}

describe('withToggler', () => {
  it('exists', () => {
    expect(withToggler).toBeDefined()
  })

  it('renders default', () => {
    const Component = withToggler(Toggler)(BaseComponent)
    const wrapper = shallow(<Component />)

    expect(wrapper).toMatchSnapshot()
  })

  it('shows base componente when visible', () => {
    const Component = withToggler(Toggler)(BaseComponent)
    const wrapper = shallow(<Component />)

    expect(wrapper).toMatchSnapshot()

    wrapper.instance().handleToggleVisibility()
    wrapper.update()

    expect(wrapper).toMatchSnapshot()

    wrapper.instance().handleToggleVisibility()
    wrapper.update()

    expect(wrapper).toMatchSnapshot()
  })
})
