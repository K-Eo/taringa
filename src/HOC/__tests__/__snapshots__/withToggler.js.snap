// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withToggler renders default 1`] = `
<React.Fragment>
  <Toggler
    isVisible={false}
    onToggleVisibility={[Function]}
  />
</React.Fragment>
`;

exports[`withToggler shows base componente when visible 1`] = `
<React.Fragment>
  <Toggler
    isVisible={false}
    onToggleVisibility={[Function]}
  />
</React.Fragment>
`;

exports[`withToggler shows base componente when visible 2`] = `
<React.Fragment>
  <Toggler
    isVisible={true}
    onToggleVisibility={[Function]}
  />
  <BaseComponent />
</React.Fragment>
`;

exports[`withToggler shows base componente when visible 3`] = `
<React.Fragment>
  <Toggler
    isVisible={false}
    onToggleVisibility={[Function]}
  />
</React.Fragment>
`;
